modular-particle-system/src/shapes/rectangle.ts

```diff
@@ -19,6 +19,49 @@ export const rectangleLogic: ShapeLogicImplementation<Rectangle> = {
         };
     },
 
+    /**
+     * Get random position on the Shape edges
+     * @returns Random position on the Shape edges
+     */
+    getRandomEdgePosition: function (shape: Rectangle): Position {
+        // NOTE: This implementation does not achieve uniform random distribution!
```

Comment:
Left a note here, but this can be merged as is. We can keep things simple.

Uniform random distribution means that it is as likely for a position to be spawned at any location along the rectangle edge.

Currently the distribution is non-uniform between the sides of the rectangle when width !== height

Reply:
Implementing uniform random distribution would be done by taking the length of rectangle sides into account (longer sides should be weighed to be favoured over shorter ones)